jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CredentialWriter.kt

```diff
@@ -21,30 +21,43 @@ import java.io.File
 
 class CreateOrUpdateCredentialProfilesAction @TestOnly constructor(
     private val writer: CredentialFileWriter,
-    private val file: File
+    private val configFile: File,
+    private val credentialsFile: File
 ) : AnActionWrapper(message("configure.toolkit.upsert_credentials.action")), DumbAware {
     @Suppress("unused")
-    constructor() : this(DefaultCredentialFileWriter, ProfileFileLocation.configurationFilePath().toFile())
+    constructor() : this(
+        DefaultConfigFileWriter,
+        ProfileFileLocation.configurationFilePath().toFile(),
+        ProfileFileLocation.credentialsFilePath().toFile()
+    )
 
     private val localFileSystem = LocalFileSystem.getInstance()
 
     override fun doActionPerformed(e: AnActionEvent) {
         val project = e.getRequiredData(PlatformDataKeys.PROJECT)
 
-        if (!file.exists()) {
-            if (confirm(project, file)) {
-                writer.createFile(file)
+        // if both config and credential files do not exist, create a new config file
+        if (!configFile.exists() && !credentialsFile.exists()) {
+            if (confirm(project, configFile)) {
+                writer.createFile(configFile)
             } else {
                 return
             }
         }
 
-        val virtualFile = localFileSystem.refreshAndFindFileByIoFile(file) ?: throw RuntimeException(message("credentials.could_not_open", file))
+        // open both config and credential files, if they exist
+        // credential file is opened last since it takes precedence over the config file
+        val virtualFiles = listOf(configFile, credentialsFile).filter { it.exists() }.map {
+            localFileSystem.refreshAndFindFileByIoFile(it) ?: throw RuntimeException(message("credentials.could_not_open", it))
+        }
+
         val fileEditorManager = FileEditorManager.getInstance(project)
 
-        localFileSystem.refreshFiles(listOf(virtualFile), false, false) {
-            fileEditorManager.openTextEditor(OpenFileDescriptor(project, virtualFile), true)
-                ?: throw RuntimeException(message("credentials.could_not_open", file))
+        localFileSystem.refreshFiles(virtualFiles, false, false) {
```

Comment:
Do we need this refresh too? We refreshed it on line 51

Reply:
Unsure; copying what Kyle wrote, but keeping it like this has the added bonus of running lines 57-63 on EDT for us, which we need to do anyways.